wavedump/src/fit-histograms

```diff
@@ -0,0 +1,566 @@
+    `l` is the mean number of primary PEs.
+    `ps` is the probability that a primary PE causes a secondary PE.
+    See this paper for a more detailed explanation:
+    https://arxiv.org/pdf/2106.13168.pdf
+    """
+    model = 0
+    for i in range(N+1):
+        model += B_coeff(i, N) * (l*(1-ps))**i * ps**(N-i)
+    model *= np.exp(-l)
+    model /= fac(N)
+    return model
+
+def B_coeff(i, N):
+    """
+    Helper function for the vinogradov model.
+    """
+    if i == 0 and N == 0:
+        return 1
+    elif i == 0 and N > 0:
+        return 0
+    else:
+        return (fac(N)*fac(N-1)) / (fac(i)*fac(i-1)*fac(N-i))
+
+class vinogradov_model:
+    """
+    p[0]: overall scale
+    p[1]: offset
+    p[2]: mean number of SPEs in integration window
+    p[3]: SPE charge
+    p[4]: std. of scope/digitizer noise
+    p[5]: std. of SPE charge
+    p[6]: probability that a primary PE triggers a secondary PE (https://arxiv.org/pdf/2106.13168.pdf)
+    """
+    def __call__(self, x, p):
+        model = 0
+        for i in range(n):
+            model += vinogradov(i, p[2], p[6]) * TMath.Gaus(x[0]-p[1], i*p[3], TMath.Sqrt(p[4]**2 + i*(p[5]**2)), False)
+        model *= p[0]
+        return model
+
+def plot_dists():
+    """
+    Plots the poisson and vinogradov distributions, then exits on user input.
+    """
+    l = 0.5
+    ps = 0.9
+    x = np.arange(20)
+    y_p = [np.exp(-l) * (l**i)/fac(i) for i in x]
+    y_v = [vinogradov(i, l, ps) for i in x]
+    print(f'poisson sum: {np.sum(y_p)}')
+    print(f'vino sum: {np.sum(y_v)}')
+    plt.figure()
+    plt.bar(x-0.2, y_p, width=0.4, label='poisson')
+    plt.bar(x+0.2, y_v, width=0.4, label='vino')
+    plt.legend()
+    plt.show()
+    input()
+    exit()
+
+if __name__ == '__main__':
+    from argparse import ArgumentParser
+    import matplotlib.pyplot as plt
+
+    parser = ArgumentParser(description='Fit SPE and sodium charge histograms')
+    parser.add_argument('filenames',nargs='+',help='input filenames (ROOT format)')
+    parser.add_argument('--sodium', default=False, action='store_true', help='flag to indicate data is from a sodium source')
+    parser.add_argument('--plot', default=False, action='store_true', help='plot the waveforms and charge integral')
+    parser.add_argument('--BV', type=int,default=42, help='Bias Voltage input')
+    #parser.add_argument('--peakper', type=double, default=0, help='Direct change of maximum percentage change from maximum peak')
+    parser.add_argument('--pdf', default=False, action='store_true', help='Save the fit canvas as pdf')
+    parser.add_argument('--ObjKeep', type=int,default=2,  help='Number of objects to fit')
+    # parser.add_argument('--SPE_FLL', default=-0.1, type=float, help='Left limit on the spe fit')
+    # parser.add_argument('--SPE_FLR', default=-0.1, type=float, help='Right limit on the spe fit')
+    parser.add_argument('--NA_GF', default=50, type=float, help='Right limit on the spe fit')
```

Comment:
lowercase. also what right limit is it referring to? What are the units?

Reply:
Sorry for leaving unclear code, again it was here before I started working so I was skeptical to alter it. Looks like it doesn't get used, so I'll delete it.